### DISCUSSION TITLE
Source structure

### DISCUSSION THREAD

**Discussion by ilemus** [freshframework/fresh#390] [Ideas]
I love the framework and how lightweight it is especially when compared to clunky frameworks like ReactJS.

Islands are a great way of compartmentalizing components which need JS on the client side only.

What about components in general? Reusable formats to be used either for server side or client side?
Where should they live?

I think it would be best to have a "components" folder by default in the file structure to make it easy to have a central location to import from for either client or server side components.

```
components
islands
routes
utils
- twind.ts
```

Also for visual studio for some reason the twind intellisense requires the setup to be in a file in the root of the project, can it be moved from utils?

```
components
islands
routes
utils
twind.config.ts
```

I manually moved it for my project, but it would be convenient if that were the default for every project.

**Comment by raymclee** [comment]
I was wondering why was the twind extension not working.
Now i know that the config file need to be at root
Thanks man!
